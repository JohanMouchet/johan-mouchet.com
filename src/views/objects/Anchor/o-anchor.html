{##
 # Anchor
 #
 # @param {array}  context                      - Used to get _SITE['rurl'] for smooth-scroll
 # @param {string} tag                          - Element tag (ex: h2)
 # @param {string} id                           - Element id for anchor reference
 # @param {string} text                         - Element text
 # @param {string} [marker : "#"]               - Anchor marker
 # @param {string} [metadata.contextClass]      - Anchor context class
 # @param {object} [metadata.attributes]        - Element attributes
 #}

{% macro default(context, options) %}
	{% set contextClass = options.metadata.contextClass %}
	{% set attributes = options.metadata.attributes %}

	<{{ options.tag }} id="{{ options.id }}" class="{{ contextClass }} u-anchor" {{ attributes|attributes|raw }}>
		<a href="{{ context._SITE['rurl'] }}#{{ options.id }}" class="{{ 'u-anchor__link'|contextClass(contextClass) }}">{{ options.marker|default('#') }}</a>
		{{- options.text|raw -}}
	</{{ options.tag }}>
{% endmacro %}
