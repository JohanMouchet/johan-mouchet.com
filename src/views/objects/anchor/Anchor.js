// @flow

import * as React from "react";
import cx from "classnames";

type Props = {
  tag: string,
  url: string,
  id: string,
  marker: string,
  className?: string | Array | Object,
  attributes?: { [key: string]: any },
  children?: Node
};

const Anchor = (props: Props) => {
  const { tag, url, id, marker, className, attributes, children } = props;

  const AnchorClass = cx("u-anchor", className);

  const Tag = tag;

  return (
    id && (
      <Tag id={id} className={AnchorClass} {...attributes}>
        <a href={url + "#" + id} className="u-anchor__link">
          {marker}
        </a>
        {children}
      </Tag>
    )
  );
};

Anchor.defaultProps = {
  tag: "span",
  marker: "#"
};

export default Anchor;
