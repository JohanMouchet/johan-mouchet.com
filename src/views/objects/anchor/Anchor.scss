.o-anchor {
  position: relative;

  /** Link **/
  &__link {
    position: absolute;
    right: 100%;
    padding-right: 0.25em;
    font-style: normal;
    font-weight: 400;
    opacity: 0;

    &:focus {
      text-decoration: none;
      opacity: 1;
    }
  }

  &:hover & {
    &__link {
      text-decoration: none;
      opacity: 1;
    }
  }
}
