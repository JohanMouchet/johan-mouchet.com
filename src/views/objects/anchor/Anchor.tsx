import * as React from "react";
import cx from "classnames";
import "./Anchor.scss";

type Props = {
  id: string;
  marker?: string;
  className?: string | string[] | { [key: string]: boolean };
  attributes?: boolean | number | string;
};

const Anchor: React.FC<Props> = ({
  id,
  marker = "#",
  className,
  ...attributes
}) => {
  const classNames = cx("o-anchor", className);

  return !id ? null : (
    <a
      className={classNames}
      href={`#${id}`}
      id={id}
      aria-hidden
      {...attributes}
    >
      {marker}
    </a>
  );
};

export default Anchor;
