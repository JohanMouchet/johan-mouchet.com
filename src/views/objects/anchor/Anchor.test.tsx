import React from "react";
import { render } from "@testing-library/react";
import Anchor from ".";

it("Renders <Anchor /> without crashing", () => {
  expect(
    render(
      <Anchor as="h2" id="id" marker="#">
        children
      </Anchor>
    ).container.firstChild
  ).toMatchSnapshot();
});
