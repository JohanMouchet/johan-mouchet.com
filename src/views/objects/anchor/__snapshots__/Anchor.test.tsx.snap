// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Renders <Anchor /> without crashing 1`] = `
<h2
  class="o-anchor"
>
  <a
    aria-label="Permalink for children"
    class="o-anchor__link"
    href="#id"
    id="id"
  >
    <span
      aria-hidden="true"
    >
      #
    </span>
  </a>
  children
</h2>
`;
